src/expense.ts

```diff
@@ -0,0 +1,98 @@
+import { collection, doc, addDoc, deleteDoc, updateDoc, Timestamp } from 'firebase/firestore';
+
+import db from './firebase';
+
+export interface ExpenseDataFirestore {
+  title: string;
+  date: Timestamp;
+  amount: number;
+}
+
+export interface ExpenseDataForm {
+  title: string;
+  date: Date;
+  amount: string;
+}
+
+export class Expense {
```

Comment:
does expense `implements` the `ExpenseDataForm`? we might have to rename the latter.

Reply:
Not sure if I follow. `ExpenseDataForm` get passed to `Expense.add()`, which adds the expense data to firestore. Essentially, it represents the raw expense data as entered into the "Add expense" form.

Reply:
i was wondering if the expense class itself also fulfills the interface ExpenseDataFrom. currently, it does, but i am not sure if this is a "coincidence" or strongly required.

[implements](https://www.typescriptlang.org/docs/handbook/2/classes.html#implements-clauses)

Reply:
Ah, okay. If I understand correctly — no, the `Expense` class does not implement `ExpenseDataForm`. `Expense` stores `amount` as a number, whereas `ExpenseDataForm` expects a string.

Reply:
ah cool. maybe add a comment in the code to clarify this subtle difference